### PR TITLE
Added test cases for lrucache, checkers, intern, hash, map and sort

### PR DESCRIPTION
**Pull request by tariq1890**

### CODE REVIEW

File: strings/intern_test.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+func (i *InternStringSuite) SetUpTest(c *C) {
+	i.pool = NewInternStringPool()
+}
+
+var _ = Suite(&InternStringSuite{})
+
+func (i *InternStringSuite) TestIntern(c *C) {
+	// Takes a string as parameter and stores it to the string pool
+	i.pool.Intern("foo string")
+	r, ok := i.pool.get("foo string")
```
Comment (pattyshack) [inline strings/intern_test.go]: test should avoid using pool's internal api?

File: strings/intern_test.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	c.Assert(ok, Equals, true)
+	c.Assert(r, Equals, "foo string")
```
Comment (pattyshack) [inline strings/intern_test.go]: Could you verify two copies of the same intern string have to the same Data pointer and Len (i.e., reflect.StringHeader are identical)?  Conversely, two copies of the same non-intern string should not have the same Data pointer.

Comment (tariq1890) [reply strings/intern_test.go]: Hello. I tried this, but even for non-intern strings the Data of stringheader end up being identical.

Comment (pattyshack) [reply strings/intern_test.go]: the golang compiler also performs string intern-ing for simple constants.  e.g., in https://play.golang.org/p/qSEMMzPZrMk y and z shares the same Data, but not x

Comment (pattyshack) [reply strings/intern_test.go]: there was a typo.  this is the corrected example https://play.golang.org/p/R5njXXk30zk

Comment (tariq1890) [reply strings/intern_test.go]: Thank you! This was helpful. I've tried various different methods of string generation and I am getting strings with distinct with stringheader data now :). strings.join, bytes array and concatenation are all generating different Data representations for the string.

**Comment by tariq1890** [comment]
@pattyshack I have addressed the review comments. Thanks :)

**Pull request merged**